libraries/stdlib/jdk7/test/PathExtensionsTest.kt

```diff
@@ -0,0 +1,353 @@
+            val child = dstFile.resolve("child")
+            Files.createFile(child)
+            assertFailsWith(DirectoryNotEmptyException::class, "copy with overwrite do not overwrite non-empty dir") {
+                srcFile.copyTo(dstFile, overwrite = true)
+            }
+            Files.delete(child)
+
+            srcFile.copyTo(dstFile, overwrite = true)
+            assertEquals(srcFile.readText(), dstFile.readText(), "copy with overwrite over empty dir")
+
+            assertTrue(Files.deleteIfExists(srcFile))
+            assertTrue(Files.deleteIfExists(dstFile))
+
+            assertFailsWith(NoSuchFileException::class) {
+                srcFile.copyTo(dstFile)
+            }
+
+            Files.createDirectory(srcFile)
+            srcFile.resolve("somefile").writeText("some content")
+            dstFile.writeText("")
+            assertFailsWith(FileAlreadyExistsException::class, "copy dir do not overwrite file") {
+                srcFile.copyTo(dstFile)
+            }
+            srcFile.copyTo(dstFile, overwrite = true)
+            assertTrue(dstFile.isDirectory())
+            assertTrue(dstFile.listDirectoryEntries().isEmpty(), "only directory is copied, but not its content")
+
+            assertFailsWith(FileAlreadyExistsException::class, "copy dir do not overwrite dir") {
+                srcFile.copyTo(dstFile)
+            }
+
+            srcFile.copyTo(dstFile, overwrite = true)
+            assertTrue(dstFile.isDirectory())
+            assertTrue(dstFile.listDirectoryEntries().isEmpty(), "only directory is copied, but not its content")
+
+            dstFile.resolve("somefile2").writeText("some content2")
+            assertFailsWith(FileAlreadyExistsException::class, "copy dir do not overwrite non-empty dir") {
+                srcFile.copyTo(dstFile, overwrite = true)
+            }
+        } finally {
+            srcFile.toFile().deleteRecursively()
+            dstFile.toFile().deleteRecursively()
+        }
+    }
+
+    @Test
+    fun copyToNameWithoutParent() {
+        val currentDir = Paths.get("").toAbsolutePath()
+        val srcFile = Files.createTempFile(null, null)
+        val dstFile = Files.createTempFile(currentDir, null, null)
+        try {
+            srcFile.writeText("Hello, World!", Charsets.UTF_8)
+            Files.delete(dstFile)
+
+            val dstRelative = Paths.get(dstFile.fileName.toString())
+
+            srcFile.copyTo(dstRelative)
+
+            assertEquals(srcFile.readText(), dstFile.readText())
+        } finally {
+            Files.delete(dstFile)
+            Files.delete(srcFile)
+        }
+    }
+
+    private fun compareFiles(src: Path, dst: Path, message: String? = null) {
+        assertTrue(dst.exists())
+        assertEquals(src.isRegularFile(), dst.isRegularFile(), message)
+        if (dst.isRegularFile()) {
+            assertTrue(src.readBytes().contentEquals(dst.readBytes()), message)
+        }
+    }
+
+    @Test
+    fun testBufferedReader() {
```

Comment:
That definitely makes more sense, I'll move them. I put them here originally because the equivalent tests for `File` live in the `File.kt` test rather than the `ReadWrite.kt` test.